### PR TITLE
Fix chi-squared tests in random_test.py

### PR DESCRIPTION
**Pull request by shoyer**
As far as I can tell, the previous implementation of the chi-squared test
for samples from discrete probability distributions was broken. It should have
been asserting that the p-value was greater 0.01, e.g., as illustrated here:
http://hamelg.blogspot.com/2015/11/python-for-data-analysis-part-25-chi.html

This hid a few other bugs, such a miscalculation of expected frequencies.

Fortunately, the existing random tests for Bernoulli and Categorical *mostly*
still pass, which the exception of multi-dimensional logits for Categorical.
Those tests are disabled by this PR.

### CODE REVIEW

**Comment by shoyer** [comment]
To be clear, I think the multidimesional categorical samplers are _probably_ generating correct values, but the tests aren't doing the right data manipulations to verify this right now.

**Review by mattjj — APPROVED** [review]
LGTM! Great attention to detail and careful fix. Thanks for catching.
